src/lib/server/server.hpp

```diff
@@ -3,23 +3,31 @@
 #include <boost/asio/io_service.hpp>
 #include <boost/asio/ip/tcp.hpp>
 
-#include "server_session.hpp"
+#include "session.hpp"
 
 namespace opossum {
 
+// The server class is responsible for setting up and listening to the system port.
 class Server {
```

Comment:
I'm missing a short description of what components are part of the server and how they interact. This would be a good place to put that.